onadata/libs/permissions.py

```diff
@@ -369,13 +426,14 @@ def _get_users_with_perms(obj, attach_perms=False, with_group_users=None):
     Returns a list of users with their permissions on an object obj.
     """
     if isinstance(obj, XForm):
-        Klass = XFormUserObjectPermission  # pylint: disable=C0103
+        user_obj_perms = XFormUserObjectPermission.objects.filter(
```

Comment:
alright, now this is getting out of hand, let's make a helper to replace lines 392-397 and 428-433:

```python
def _get_user_obj_perms(obj):
    if isinstance(obj, XForm):
        klass = XFormUserObjectPermission
    elif isinstance(obj, Project):
        klass = ProjectUserObjectPermission
    
    return klass and klass.objects.filter(content_object_id=obj.pk)
```

then use as:

```python
def _get_users_with_perms(obj, attach_perms=False, with_group_users=None):
    user_obj_perms = _get_user_obj_perms(obj)

    if not user_obj_perms:
                 return get_users_with_perms(obj, attach_perms=attach_perms,
                                     with_group_users=with_group_users)
    ...
```

Reply:
There is a difference, the models are different one affects the group and the other affects the user.